### ISSUE TITLE
Normalizing batches without control samples

### ISSUE THREAD

**Issue by algebio** [MarioniLab/cydar#4]
Hi

I have a dataset with 19 batches but some of them don´t include a control sample, I also have fcs files from the Fluidigm calibration beads from all the samples.

Is there any way to reduce the batch effect of my dataset like using the Fluidgm beads or the peak signals or from the batches which do have control samples or anything else? I don´t want to discard those batches but I can´t use them without correcting the batch effect.

Thank you for your help.

Regards
Juan

**Comment by LTLA** [comment]
We don't do much with the calibration beads. From what I understand, the Fluidigm software already adjusts that for you, and you can check this by plotting the signal from the beads (or any other marker) over time - you should see a flat line, and that's what I typically see in my FCS files. If you don't see that, I believe that the [**CATALYST** package](http://bioconductor.org/packages/release/bioc/html/CATALYST.html) provides some utilities for using the control beads to correct for the loss of sensitivity over time, but that's independent of anything that **cydar** does.

As for your problem, there are various possible approaches. Even if you do not have "technical control samples" in each batch, perhaps the experiment was still sufficiently well-designed that you have a common treatment group across all batches, e.g., all batches have samples from normal patients. In that case, you can use this common group to do your batch correction via `normalizeBatches()`. 

If you don't have some kind of common group across batches, then life becomes more difficult and you would need to make some assumptions. For example, you could apply `normalizeBatches()` without specifying the treatment structure within each batch; the assumption here is that any biological differences cancel out when averaging samples within each batch, such that any remaining differences between batches are purely technical in origin. 

Alternatively, you might run `batchelor::fastMNN()` on each batch, which assumes that (i) the batches share at least a few subpopulations and (ii) any differences in marker expression between batches for those shared populations are not of interest, biological or otherwise.

If none of this is possible or palatable, you may consider performing analyses within each batch (assuming that you have enough replicates per batch) and then combining the inferences in a meta-analysis. This requires that you gather per-batch results in a way that is easily compared across batches, and I'll admit that **cydar** is not the best choice for that. If you go down this route, I would use gating to obtain cell populations that can be linked across batches.

**Comment by algebio** [comment]
Thank you so much for your extended answer, it's so good when authors are so helpful. Definitively it's an added value to Cydar.

We're studying minimal residual disease in AML. My doubt is this: Taking into account the two assumptions of batchelor::fastMNN(). Could I use it without the risk of loosing information about the expression of shared important subpopulations?

Regards
Juan

**Comment by LTLA** [comment]
> Could I use it without the risk of loosing information about the expression of shared important subpopulations?

Depends on what you're expecting. 

The batch correction will yield intensities where the differences between batches are removed. If one batch is disease and another batch is control, the method can't tell whether those differences are technical (e.g., differences in antibody staining concentration) or biological (some effect of disease). It'll just remove them all; and yes, this is desired behavior, see the discussion [here](https://osca.bioconductor.org/multi-sample-comparisons.html#sacrificing-differences). 

In short, the batch-corrected intensities are only safe to use for identifying common subpopulations, e.g., by clustering or gating. However, once you have those subpopulations, you can always go back to the original intensities to compute aggregate values (usually the median non-corrected intensity across all cells for each marker) for differential analyses between conditions. This will recover the differences that were intentionally discarded by the batch correction algorithm.

**Comment by algebio** [comment]
Sorry Aaron, I forgot to thank you for all your help. Really appreciated.

Regards
Juan
